Utilities/rANS/include/rANS/FrequencyTable.h

```diff
@@ -173,87 +186,39 @@ void FrequencyTable::addFrequencies(Freq_IT begin, Freq_IT end, value_t min, val
   LOG(trace) << "done adding frequencies";
 }
 
-inline typename FrequencyTable::value_t FrequencyTable::operator[](value_t index) const
-{
-  const value_t idx = index - mMin;
-  assert(idx >= 0);
-  assert(idx < mFrequencyTable.size());
-  return mFrequencyTable[index];
-}
-
-inline size_t FrequencyTable::size() const
-{
-  return mFrequencyTable.size();
-}
-
-inline const uint32_t* FrequencyTable::data() const
-{
-  return mFrequencyTable.data();
-}
-
-inline const uint32_t* FrequencyTable::cbegin() const
-{
-  return mFrequencyTable.data();
-}
-
-inline const uint32_t* FrequencyTable::cend() const
-{
-  return mFrequencyTable.data() + mFrequencyTable.size();
-}
-
-inline uint32_t* FrequencyTable::begin()
-{
-  return const_cast<uint32_t*>(static_cast<const FrequencyTable*>(this)->begin());
-}
-
-inline uint32_t* FrequencyTable::end()
-{
-  return const_cast<uint32_t*>(static_cast<const FrequencyTable*>(this)->end());
-}
-
-inline const uint32_t* FrequencyTable::begin() const
-{
-  return cbegin();
-}
-
-inline const uint32_t* FrequencyTable::end() const
-{
-  return cend();
-}
-
 inline FrequencyTable& FrequencyTable::operator+(FrequencyTable& other)
 {
-  addFrequencies(std::begin(other), std::end(other), other.getMinSymbol(), other.getMaxSymbol());
+  addFrequencies(other.cbegin(), other.cend(), other.getMinSymbol(), other.getMaxSymbol());
   return *this;
 }
 
-inline typename FrequencyTable::value_t FrequencyTable::getMinSymbol() const
+inline auto FrequencyTable::release() && noexcept -> histogram_t&&
 {
-  return mMin;
-}
+  mMin = 0;
+  mMax = 0;
+  mNumSamples = 0;
+  return std::move(mFrequencyTable);
+};
 
-inline typename FrequencyTable::value_t FrequencyTable::getMaxSymbol() const
+inline size_t FrequencyTable::getNUsedAlphabetSymbols() const noexcept
 {
-  return mMax;
+  return std::count_if(cbegin(), cend(), [](count_t count) { return count > 0; });
 }
 
-inline size_t FrequencyTable::getAlphabetRangeBits() const
+inline auto FrequencyTable::getSymbol(symbol_t symbol) const -> const count_t&
 {
-  if (mMax - mMin > 0) {
-    return std::ceil(std::log2(mMax - mMin));
-  } else if (mMax - mMin == 0) {
-    return 1;
-  } else {
-    return 0;
-  }
+  //negative numbers cause overflow thus we get away with one comparison only
+  const size_t index = static_cast<size_t>(symbol - mMin);
+  assert(index < mFrequencyTable.size());
+  return mFrequencyTable[index];
 }
 
-inline size_t FrequencyTable::getNumSamples() const
+inline auto FrequencyTable::getSymbol(symbol_t symbol) -> count_t&
 {
-  return mNumSamples;
+  return const_cast<count_t&>(static_cast<const FrequencyTable&>(*this).getSymbol(symbol));
 }
 
-}  // namespace rans
-}  // namespace o2
+} // namespace rans
+} // namespace o2
```

Comment:
For the record: you can actually use:

```
namespace o2::framework {
}
```

now...

Reply:
Again something new I learned. thanks :)